### PR TITLE
System Events charts period filter limits

### LINKED ISSUE #51: Set max/min values when changing the period filter
This would help align the timeline for system events charts in particular because they can go for periods without any events, and so the chart picks up a smaller range than the one set via the filter, and often different between the charts

### PR DESCRIPTION
**Pull request by ktkization**
closes #51

### CODE REVIEW

File: webapp/lib/charts.dart [hunk 1]
```diff
@@ -249,7 +249,7 @@ class SystemEventsTimeseriesLineChartView {
     chart = chartjs.Chart(canvas.getContext('2d'), chartConfig);
   }
 
-  void updateChart(Map<String, Map<DateTime, num>> updatedCountsAtTimestampList, {String timeScaleUnit = 'day', num upperLimit}) {
+  void updateChart(Map<String, Map<DateTime, num>> updatedCountsAtTimestampList, {String timeScaleUnit = 'day', num upperLimit, Map<String, DateTime> periodLimits}) {
```
Comment (marianamarasoiu) [inline webapp/lib/charts.dart]: I think I'd prefer passing in the limits via separate variables, same as we do for `upperLimit`. I suggest `xLowerLimit` and `xUpperLimit`.

We'll probably want to transform `upperLimit` into `yLowerLimit` and `yUpperLimit` in a follow up PR as well

File: webapp/lib/controller.dart [hunk 2]
```diff
@@ -413,14 +413,29 @@ void updateSystemEventsCharts(List<model.SystemEventsData> filteredSystemEventsD
...
 
+  var now = new DateTime.now();
+  
+  Map<ChartPeriodFilters, DateTime> periodFiltersDateMap = {
+    ChartPeriodFilters.alltime: null,
+    ChartPeriodFilters.days1: new DateTime(now.year, now.month, now.day - 1, 00),
+    ChartPeriodFilters.days8: new DateTime(now.year, now.month, now.day - 8, 00),
+    ChartPeriodFilters.days15: new DateTime(now.year, now.month, now.day - 15, 00),
+    ChartPeriodFilters.month1: new DateTime(now.year, now.month - 1 , now.day, 00)
+  };
```
Comment (marianamarasoiu) [inline webapp/lib/controller.dart]: Consider taking this out and turn it into a helper function. We might want to use in the future for the other charts as well (and we probably should!)

Something like
```
DateTime getStartDateTimeForPeriod(ChartPeriodFilters period) {
  switch (period) {
    ...
  }
}
```

**Review by marianamarasoiu — APPROVED** [review]
thanks @ktkization - a couple of suggestions to make the code a bit more reusable

File: webapp/lib/charts.dart (hunk 1 above)
Comment (ktkization) [reply webapp/lib/charts.dart]: I've created the separate variables `xLowerLimit` and  `xUpperLimit`.

I'll change the previous `upperLimit` into `yLowerLimit` and `yUpperLimit` in a follow-up PR.

File: webapp/lib/controller.dart (hunk 2 above)
Comment (ktkization) [reply webapp/lib/controller.dart]: Made this into a helper function as suggested

**Review by marianamarasoiu — APPROVED** [review]

**Comment by marianamarasoiu** [comment]
sorry for the delay! looks great, thanks @ktkization !

**Pull request closed without merging**